feat(evm-bridge): add native token evm bridge tests

brancoder opened the pull request:
# Description of change

adds tests for native tokens 
resolves #7708 

## Links to any relevant issues

Be sure to reference any related issues by adding `fixes #(issue)`.

## How the change has been tested

- [ ] Basic tests (linting, compilation, formatting, unit/integration tests)
- [ ] Patch-specific tests (correctness, functionality coverage)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

brancoder committed:
feat: add fund L1 adddress with native tokens

brancoder committed:
feat: update fundL2AddressWithIscClient method to send native tokens

brancoder committed:
feat: add checkL2CoinBalanceForAddress util method

brancoder committed:
add data-testid and tool coin type to apps backend

brancoder committed:
feat: add test to bridge max native tokens L1->L2

brancoder committed:
fix: improve util methods for funding and fetching balances

brancoder committed:
feat: add bridge max native tokens L2=>L1

brancoder committed:
fix: add refetch interval to get l2 balances

brancoder committed:
fix: remove extra constant TOKEN_COIN_TYPE

brancoder committed:
refactor: refactor deposit than withdraw tests

brancoder committed:
fix: add balance checks with retries when fetching all balances

brancoder committed:
feat: add send native tokens test to roundtrip tests

msarcev commented:
Would prefer to have new files with the added tests, instead of making the current two specs longer with `serial`.
`serial` makes the tests take longer (no parallel execution), and if one "step" fails in the chain we don't get to see the results of the rest.

brancoder commented:
> Would prefer to have new files with the added tests, instead of making the current two specs longer with `serial`. `serial` makes the tests take longer (no parallel execution), and if one "step" fails in the chain we don't get to see the results of the rest.

The tests are set up this way to keep wallet creations and faucet requests to a minimum, so we don't hit the faucet limit.

brancoder committed:
feat: add L2=>L1 native token test to roundtrip tests

brancoder committed:
fix: split max tests to dedicated files

brancoder committed:
fix: split roundtrip tests to dedicated files

brancoder committed:
fix: typo

brancoder committed:
Merge branch 'develop' into tooling/add-native-token-bridge-tests

brancoder committed:
fix: typo

brancoder committed:
fix: typo

brancoder committed:
refactor(evm-bridge): separate utils from helpers for improved test maintainability (#7871)

# Description of change

resolves #7870

 Improve E2E test reliability and maintenance #7876
- Improve browser context management and clean teardown
- Extract wallet setup steps to reusable utility functions
- Extract UI steps to reusable utility functions
- Add global setup step that runs before all tests and distributes funds to test addresses
- Add robust waiting mechanism

brancoder committed:
Merge branch 'develop' into tooling/add-native-token-bridge-tests

msarcev committed:
Merge branch 'develop' into tooling/add-native-token-bridge-tests

panteleymonchuk committed:
Merge remote-tracking branch 'origin/develop' into tooling/add-native-token-bridge-tests

panteleymonchuk committed:
fix: playwright

panteleymonchuk committed:
chore: update imports

panteleymonchuk committed:
Merge remote-tracking branch 'origin/develop' into tooling/add-native-token-bridge-tests

brancoder committed:
Merge branch 'develop' into tooling/add-native-token-bridge-tests

brancoder committed:
fix: comments and add wait for l1 address to connects to fixture

brancoder committed:
Merge branch 'develop' into tooling/add-native-token-bridge-tests

brancoder committed:
fix: reorder waiting time

marc2332 committed:
Merge branch 'develop' into tooling/add-native-token-bridge-tests

VmMad committed:
Merge branch 'develop' into tooling/add-native-token-bridge-tests

msarcev committed:
Merge branch 'develop' into tooling/add-native-token-bridge-tests
